src/terminal/Grid.cpp

```diff
@@ -123,9 +125,22 @@ template <typename Cell>
 Grid<Cell>::Grid(PageSize _pageSize, bool _reflowOnResize, LineCount _maxHistoryLineCount):
```

Comment:
```suggestion
Grid<Cell>::Grid(PageSize _pageSize, bool _reflowOnResize, MaxHistoryLineCount _maxHistoryLineCount):
```
I think it would be best to also adapt the calling parameters to strongly type the max history line count. This way you enable the caller to explicitly pass `Infinite {}` (which btw should not be a child struct of `Grid<T>` in order to easy usage of that type. Just move this empty struct to `<terminal/primitives.h>`, somewhere around `LineCount`, same for `MaxHistoryLineCount`, such that you can even make use of `MaxHistoryLineCount` in the `Config` struct to properly reflect infinite scroll history state in a typed way (rather than knowing that -1 means infinite).

Reply:
> I think it would be best to also adapt the calling parameters to strongly type the max history line count. This way you enable the caller to explicitly pass `Infinite {}` (which btw should not be a child struct of `Grid<T>` in order to easy usage of that type. Just move this empty struct to `<terminal/primitives.h>`, somewhere around `LineCount`, same for `MaxHistoryLineCount`, such that you can even make use of `MaxHistoryLineCount` in the `Config` struct to properly reflect infinite scroll history state in a typed way (rather than knowing that -1 means infinite).

Changed it, see 'Refactor' commit